File: CsharpTranslationFiesta/Logger.cs
using System;
using System.IO;

namespace CsharpTranslationFiesta
{
    public static class Logger
    {
        private static readonly object _sync = new object();
        private static readonly string LogFilePath = Path.Combine(AppContext.BaseDirectory, "csharptranslationfiesta.log");

        public static void Info(string message) => Write("INFO", message);
        public static void Error(string message) => Write("ERROR", message);
        public static void Debug(string message) => Write("DEBUG", message);
        public static void Warn(string message) => Write("WARN", message);

        private static void Write(string level, string message)
        {
            try
            {
                lock (_sync)
                {
                    File.AppendAllText(LogFilePath, $"[{DateTime.UtcNow:O}] {level}: {message}{Environment.NewLine}");
                }
            }
            catch
            {
                // Best-effort logging: never throw from logger
            }
        }
    }
}

File: TranslationFiestaFSharp/Logger.fs
namespace TranslationFiestaFSharp

module Logger =

    open System

    let private lockObj = obj()
    let private logFile = PortablePaths.logFilePath

    let private write level message =
        try
            lock lockObj (fun () ->
                    let logEntry = sprintf "[%s] %s: %s%s" (DateTime.UtcNow.ToString("O")) level message Environment.NewLine
                    System.IO.File.AppendAllText(logFile, logEntry)
            )
        with
        | _ -> () // Best effort logging; don't crash app

    let info message = write "INFO" message
    let error message = write "ERROR" message
    let debug message = write "DEBUG" message
